cocos/core/utils/array.ts

```diff
@@ -24,7 +24,71 @@
 
 import { logID } from '../platform/debug';
 
-export { default as MutableForwardIterator } from './mutable-forward-iterator';
+/**
+ * @example
+ * ```
+ * import { js } from 'cc';
+ * var array = [0, 1, 2, 3, 4];
+ * var iterator = new js.array.MutableForwardIterator(array);
+ * for (iterator.i = 0; iterator.i < array.length; ++iterator.i) {
+ *     var item = array[iterator.i];
+ *     ...
+ * }
+ * ```
+ */
+export class MutableForwardIterator<T> {
+    public i = 0;
+
+    constructor (public array: T[]) {
+    }
+
+    get length (): number {
+        return this.array.length;
+    }
+
+    set length (value: number) {
+        this.array.length = value;
+        if (this.i >= value) {
+            this.i = value - 1;
+        }
+    }
+
+    public remove (value: T): void {
+        const index = this.array.indexOf(value);
+        if (index >= 0) {
+            this.removeAt(index);
+        }
+    }
+
+    public removeAt (i: number): void {
+        this.array.splice(i, 1);
+
+        if (i <= this.i) {
+            --this.i;
+        }
+    }
+
+    public fastRemove (value: T): void {
+        const index = this.array.indexOf(value);
+        if (index >= 0) {
+            this.fastRemoveAt(index);
+        }
+    }
+
+    public fastRemoveAt (i: number): void {
+        const array = this.array;
+        array[i] = array[array.length - 1];
+        --array.length;
+
+        if (i <= this.i) {
+            --this.i;
+        }
+    }
+
+    public push (item: T): void {
+        this.array.push(item);
+    }
```

Comment:
Why move the file content of `mutable-forward-iterator.ts` to here?

Reply:
When compiling the dts, this definition will be lost, so it will be normal to put it here.